src/peergos/server/util/HttpUtil.java

```diff
@@ -54,4 +55,131 @@ public static void replyError(HttpExchange exchange, Throwable t) {
             Logging.LOG().log(Level.WARNING, e.getMessage(), e);
         }
     }
+
+    public static byte[] get(PresignedUrl url) throws IOException {
+        try {
+            HttpURLConnection conn = (HttpURLConnection) new URI(url.base).toURL().openConnection();
+            conn.setRequestMethod("GET");
+            for (Map.Entry<String, String> e : url.fields.entrySet()) {
+                conn.setRequestProperty(e.getKey(), e.getValue());
+            }
+
+            try {
+                InputStream in = conn.getInputStream();
+                ByteArrayOutputStream resp = new ByteArrayOutputStream();
```

Comment:
coud refactor out the copy to BAOS

Reply:
how?

Reply:
I  mean a util method to copy from stream to byte[].  ie. byte[] bytes = Utils.copy(is);

Reply:
Ah like Serialize.readFully :-) Done.